numba/annotations/type_annotations.py

```diff
@@ -227,32 +227,23 @@ def add_ir_line(func_data, line):
                     elif line.strip().endswith('pyobject'):
                         func_data['python_tags'][num] = 'object_tag'
 
-        # If there are no lifted loops to compile, or if there are lifted loops
-        # to compiled and they've all been compiled, then write annotations
-        # for current function.
-        if ((len(self.lifted) == 0 and self.lifted_from is None) or
-                (self.lifted_from is not None and
-                 self.lifted_from[1]['num_lifted_loops'] == 0)):
-
-            # If jinja2 module is not installed we should never get here,
-            # but just in case...
-            try:
-                from jinja2 import Template
-            except ImportError:
-                raise ImportError("please install the 'jinja2' package")
-
-            root = os.path.join(os.path.dirname(__file__))
-            template_filename = os.path.join(root, 'template.html')
-            with open(template_filename, 'r') as template:
-                html = template.read()
-
-            template = Template(html)
-            rendered = template.render(func_data=TypeAnnotation.func_data)
-            if outfile is None:
-                with open(self.html_output, 'w') as output:
-                    output.write(rendered)
-            else:
-                outfile.write(rendered)
+        try:
+            from jinja2 import Template
+        except ImportError:
+            raise ImportError("please install the 'jinja2' package")
+
+        root = os.path.join(os.path.dirname(__file__))
+        template_filename = os.path.join(root, 'template.html')
+        with open(template_filename, 'r') as template:
+            html = template.read()
+
+        template = Template(html)
+        rendered = template.render(func_data=TypeAnnotation.func_data)
+        if outfile is None:
```

Comment:
Yes, it can happen.  So, I inverted the responsibility to the caller.  The method will always take a `outfile` argument now.